eng/dev_requirements.txt

```diff
@@ -1,8 +1 @@
-azure-pylint-guidelines-checker==0.0.8
-colorama==0.4.6
-debugpy==1.8.9
-pytest==8.3.4
-tox==4.23.2
-coverage==7.6.9
-black==24.10.0
-setuptools==73.0.1
+-r ../packages/typespec-python/dev_requirements.txt
```

Comment:
Although it is a little weird to install dependencies of subdirectory, we only need to update dependencies in one file after this change.